packages/webpack-plugin/src/compiler.ts

```diff
@@ -55,6 +55,11 @@ function getRootCompilation(loader: LoaderContext) {
   return compilation;
 }
 
+const templateStringRegexp = /\[([^\[\]\.]+)\]/g;
+
+export const escapeWebpackTemplateString = (s: string) =>
+  s.replaceAll(templateStringRegexp, '[\\$1\\]');
```

Comment:
I went with the `$1` syntax as my assumption is that's less overhead compared to creating a replacement function.